.generator/src/generator/templates/client.j2

```diff
@@ -115,35 +118,119 @@ func ParameterToString(obj interface{}, collectionFormat string) string {
 {# The method is used in Terraform client and needs to be public. -#}
 // CallAPI do the request.
 func (c *APIClient) CallAPI(request *http.Request) (*http.Response, error) {
-	if c.Cfg.Debug {
-		dump, err := httputil.DumpRequestOut(request, true)
-		if err != nil {
-			return nil, err
+	var ccancel context.CancelFunc
+	var rawBody []byte
+	if request.Body != nil && request.Body != http.NoBody {
+		rawBody, _ = ioutil.ReadAll(request.Body)
+		request.Body.Close()
+	}
+
+	ctx := request.Context()
+	if _, set := ctx.Deadline(); !set {
+		ctx, ccancel = context.WithTimeout(ctx, c.Cfg.RetryConfiguration.HTTPRetryTimeout)
+		defer ccancel()
+	}
+	maxRetries := c.Cfg.RetryConfiguration.MaxRetries
+	retryCount := 0
+	for {
+		if retryCount == maxRetries {
+			ccancel()
+		}
+		newRequest := copyRequest(request, &rawBody)
+		if c.Cfg.Debug {
+			dump, err := httputil.DumpRequestOut(newRequest, true)
+			if err != nil {
+				return nil, err
+			}
+			// Strip any api keys from the response being logged
+			keys, ok := newRequest.Context().Value(ContextAPIKeys).(map[string]APIKey)
+			if keys != nil && ok {
+				for _, apiKey := range keys {
+					valueRegex := regexp.MustCompile(fmt.Sprintf("(?m)%s", apiKey.Key))
+					dump = valueRegex.ReplaceAll(dump, []byte("REDACTED"))
+				}
+			}
+			log.Printf("\n%s\n", string(dump))
+		}
+		resp, requestErr := c.Cfg.HTTPClient.Do(newRequest)
+
+		if requestErr != nil {
+			return resp, requestErr
 		}
-		// Strip any api keys from the response being logged
-		keys, ok := request.Context().Value(ContextAPIKeys).(map[string]APIKey)
-		if keys != nil && ok {
-			for _, apiKey := range keys {
-				valueRegex := regexp.MustCompile(fmt.Sprintf("(?m)%s", apiKey.Key))
-				dump = valueRegex.ReplaceAll(dump, []byte("REDACTED"))
+
+		retryDuration, shouldRetry := c.shouldRetryRequest(resp, retryCount)
+		if !shouldRetry {
+			if c.Cfg.Debug {
+				dump, err := httputil.DumpResponse(resp, true)
+				if err != nil {
+					return resp, err
+				}
+				log.Printf("\n%s\n", string(dump))
+				return resp, requestErr
+			} else {
+				return resp, requestErr
 			}
```

Comment:
```suggestion
	return resp, requestErr
```

Lets log the response once on line 160 and not repeat the same block multiple times